docs/debug.md

```diff
@@ -0,0 +1,650 @@
+
+The `break` command can be used to set a breakpoint to a specific line in the
+program. An example could be `(gdb) break lala.c:2`. To start running the
+program, use the `run` command. The program will stop once it reaches that
+line and can be traced with the `step` and `next` commands. Alternatively the
+operation of the program can be resumed via the `continue` command. If the
+program crashes gdb will stop and the `backtrace` command can be used to
+inspect what happened before the program crashed.
+
+Here is an example session:
+
+```
+snappy-debug.gdb application --args additional arguments
+(gdb) run
+Program received signal SIGSEGV, Segmentation fault.
+0x000000000040053f in main () at lala.c:4
+4        printf("%i", *(int*)0);
+(gdb) backtrace
+#0  0x000000000040053f in main () at lala.c:4
+```
+
+### strace
+
+The strace tool can be used to inspect what system calls an application
+performs.
+
+The command output looks like this:
+
+```
+$ snappy-debug.strace cat /etc/shadow
+...
+open("/etc/shadow", O_RDONLY)           = -1 EACCES (Permission denied)
+...
+```
+
+This indicates that the open system call failed with the given error number.
+
+### ltrace
+
+The ltrace tool is similar to strace. The difference is that it tracks
+library calls instead of system calls.
+
+The command output looks like this:
+
+```
+$ snappy-debug.ltrace /etc/shadow
+__libc_start_main(0x401a40, 2, 0x7ffc27e9f058, 0x408b40 <unfinished ...>
+getpagesize()                                    = 4096
+strrchr("cat", '/')                              = nil
+setlocale(LC_ALL, "")                            = "en_US.UTF-8"
+bindtextdomain("coreutils", "/usr/share/locale") = "/usr/share/locale"
+textdomain("coreutils")                          = "coreutils"
+__cxa_atexit(0x402ac0, 0, 0, 0x736c6974756572)   = 0
+getopt_long(2, 0x7ffc27e9f058, "benstuvAET", 0x409280, nil) = -1
+__fxstat(1, 1, 0x7ffc27e9eea0)                   = 0
+open("/etc/shadow", 0, 0400000)                  = -1
+__errno_location()                               = 0x7f1ed2527690
+```
+
+## Important system logs
+
+Snappy Ubuntu system logs are mapped to the classic location of those logs.
+The most important log file to look at is:
+
+    /var/log/syslog
+
+The syslog is particularly useful since kernel logs, launcher output, service
+status, system logs and confinement violations (these are covered in depth
+elsewhere) all get logged there.
+
+Further, systemd has log dump facility to look at all system and snap
+services. The "snappy service" command provides an easy way to monitor and
+inspect snap services. To get the status of all snap services run "snappy
+service status". To inspect the logs run "snappy service logs". Both commands
+can take a snap name to limit to a specific snap and also `<snapname>` to
+limit to only a specific service of a snap (ie for the shout service in
+`shout.sergiusens` run: `sudo snappy service logs shout`).
```

Comment:
> Both commands can take a snap name to limit to a specific snap and also `<snapname>` to limit to only a specific service of a snap[...]

I don't quite understand what you're saying here. In my testing, there was no way to limit the logs to only a specific service of the .snap-- only `sudo snappy service logs <package name>`.

> ie for the shout service in [...]

Pedantic, but you probably want "e.g." here instead of "i.e.".